src/04-modality-specific-files/05-task-events.md

```diff
@@ -102,20 +108,41 @@ onset duration  trial_type  identifier  database  response_time
 5.6 0.6 sad AF01ANSA  kdef  1.739
 ```
 
-This file should be accompanied by a data dictionary.  
+This file should be accompanied by a data dictionary.
 
 ```Text
 sub-control01/
     func/
         sub-control01_task-emoface_events.json
 ```
 
-The `trial_type` and `identifier` columns from the `*_events.tsv` files might be described in this
-dictionary as follows. 
-Note that all other columns SHOULD also be described but are omitted for the sake
-of example.
+Specific fields related to stimulus presentation details SHOULD also be present
+in this data dictionary:
+
+| Column name                      | Description                                                                                                                                                                                                                                  |
+|----------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| StimulusPresentationOS           | RECOMMENDED. Operating system used to run the stimuli presentation software (for formatting recommendations, see examples below this table).                                                                                                 |
+| StimulusPresentationSoftwareName | RECOMMENDED. Name of the software that was used to present the stimuli                                                                                                                                                                       |
+| StimulusPresentationSoftwareRRID | RECOMMENDED. Name ideally with its [Research Resource Identifier](https://scicrunch.org/resources) of the software that was used to present the stimuli Examples: the RRID for Psychtoolbox is 'SCR_002881' that of PsychoPy is 'SCR_006571' |
+| StimulusPresentationVersion      | RECOMMENDED. Version of the software that was used to present the stimuli.                                                                                                                                                                   |
+| StimulusPresentationCode         | RECOMMENDED. DOI of the archive (for example on Zenodo, figshare...) for the code that was used to present the stimuli. Alternatively this could be a URL to code on GitHub, GitLab, etc.                                                    |
```

Comment:
Is this intended to be the code for e.g. PsychoPy ? Or for the experiment ? Personally I paused on this, and it looks like the example DOI points to BIDS, so it didn't clarify for me.

Since everything else with `StimulusPresentation` is about the infrastructure (e.g. OS and software platform), if this is intended to refer to the stimulus (or task) itself, I'd probably remove `Presentation` from this field name !

Reply:
Yes this is supposed to point to code for psychopy, psychtoolbox, expyriment, Presentation, ePrime...

Good catch I think I had assumed the DOI that was suggested before pointed to actual code: I can add the DOI to one my experiment but definitely not showcasing best practices (it is a presentation script for starter...) So if anyone has a DOI to some cleaner code, we can use that instead.

Removing `Presentation`: could do except maybe for `StimulusPresentationVersion` which would then become `StimulusVersion` and might be more understood as referring to the version of the stimulus database that was used (for a set of standardized images for example).

Alternatively...

A cleaner and more readable option might be to do what is done for the `GeneratedBy` field to describe [derived data](https://bids-specification.readthedocs.io/en/stable/03-modality-agnostic-files.html#derived-dataset-and-pipeline-description) and have StimulusPresentation present like this:

```
"StimulusPresentation": 
{
   "OperatingSystem": "Linux-Ubuntu 18.04.5 LTS",
   "SoftwareName": "Psychtoolbox",
   "SoftwareRRID": "SCR_002881",
   "SoftwareVersion": "3.0.14",
   "Code": "10.5281/zenodo.3581316"
}
```

Reply:
+1 for adopting a similar specification as for the `GeneratedBy` field!

re: "good practice" code ... I have some code here that I consider okay --> [10.5281/zenodo.3354368](https://doi.org/10.5281/zenodo.3354368)

feel free to judge yourself and pick whatever DOI you think best shows off "good practice"

Reply:
@sappelhoff @emdupre 
I made some changes: let me know what you think